stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/gallery/AttachmentGalleryDestination.kt

```diff
@@ -0,0 +1,81 @@
+package io.getstream.chat.android.ui.gallery
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.ActivityResultRegistry
+import androidx.activity.result.contract.ActivityResultContract
+import com.getstream.sdk.chat.navigation.destinations.ChatDestination
+
+public class AttachmentGalleryDestination(
+    context: Context,
+    private val attachmentReplyOptionHandler: AttachmentGalleryActivity.AttachmentReplyOptionHandler,
+    private val attachmentShowInChatOptionHandler: AttachmentGalleryActivity.AttachmentShowInChatOptionHandler,
+    private val attachmentDownloadOptionHandler: AttachmentGalleryActivity.AttachmentDownloadOptionHandler,
+    private val attachmentDeleteOptionClickHandler: AttachmentGalleryActivity.AttachmentDeleteOptionHandler,
+) : ChatDestination(context) {
+    private var launcher: ActivityResultLauncher<AttachmentGalleryResultContract.Input>? = null
+
+    private lateinit var attachmentGalleryItems: List<AttachmentGalleryItem>
+    private var attachmentIndex: Int = 0
+
+    public fun setData(attachmentGalleryItems: List<AttachmentGalleryItem>, attachmentIndex: Int) {
+        this.attachmentGalleryItems = attachmentGalleryItems
+        this.attachmentIndex = attachmentIndex
+    }
+
+    override fun navigate() {
+        if (attachmentGalleryItems.isEmpty()) {
+            Toast.makeText(context, "Invalid image(s)!", Toast.LENGTH_SHORT).show()
+            return
+        }
+
+        AttachmentGalleryRepository.setAttachmentGalleryItems(attachmentGalleryItems)
+        launcher?.launch(AttachmentGalleryResultContract.Input(attachmentIndex))
+    }
+
+    public fun register(activityResultRegistry: ActivityResultRegistry) {
+        launcher = activityResultRegistry.register("attachment_gallery_launcher", AttachmentGalleryResultContract()) {
+            it?.apply {
```

Comment:
No need for this `it?.apply` the when cases will already smart cast the `it` to non null. You can erase this line without affecting this rest of the code.

Reply:
Nope, doesn't compile without it

Reply:
Added a commit https://github.com/GetStream/stream-chat-android/pull/1316/commits/020a67e441a2a0e1f52704c99ef9ab858161fda9 that removes the extra safe cast, compiles just fine

Reply:
Strange, my mistake, sorry.